v0/destinations/hs/util.js

```diff
@@ -376,6 +379,181 @@ const getEventAndPropertiesFromConfig = (message, destination, payload) => {
   return payload;
 };
 
+/**
+ *
+ * To reduce the number of calls for searching of already existing objects
+ * We do search for all the objects before router transform and assign the type (create/update)
+ * accordingly to context.hubspotOperation
+ *
+ * */
+
+const splitEventsForCreateUpdate = async (inputs, destination) => {
+  // get all the id and properties of already existing objects needed for update.
+  const updateHubspotIds = await getExistingData(inputs, destination);
+  const resultInput = [];
+
+  inputs.map(input => {
+    const { message } = input;
+    const { destinationExternalId } = getDestinationExternalIDInfoForRetl(
+      message,
+      "HS"
+    );
+
+    let filteredInfo = updateHubspotIds.filter(
+      update => update.property === destinationExternalId
+    );
+
+    if (filteredInfo.length) {
+      input.message.context.externalId = setHsSearchId(
+        input,
+        filteredInfo[0].id
+      );
+      input.message.context.hubspotOperation = "updateObject";
+      resultInput.push(input);
+    } else {
+      input.message.context.hubspotOperation = "createObject";
```

Comment:
we shouldn't be assigning this to context rather it should be placed at the root level as we have predefined things that context should contain. cc @utsabc 
```suggestion
      input.message.hubspotOperation = "updateObject";
      resultInput.push(input);
    } else {
      input.message.hubspotOperation = "createObject";
```

Reply:
context is something related to Rudderstack info. Like we set externalId's etc don't think it be great to change the whole structure of message with a new key